Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 3] Disabling HTTPS url scheme for API(s) of all API Management services..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+    # Includes API Management services, to which, previously made changes were successfully rolled back.
+    $apiManagementServicesRolledBack = @()
+
+    # Includes API Management services that were skipped during roll back. There were errors rolling back the changes made previously.
+    $apiManagementServicesSkipped = @()
+
+
+    $validAPIManagementServiceDetails | ForEach-Object {
+        $apiManagementservice = $_
+           
+        Write-Host "Enabling HTTPS url scheme for API(s) of API Management Service: Resource ID - $($_.ResourceId), Resource Group Name - $($_.ResourceGroupName), Resource Name - $($_.ResourceName)" -ForegroundColor $([Constants]::MessageType.Warning)
+            
+        try
+        {           
+            $listOfApiRolledBack = @()
+            $listOfApiSkipped = @()
+            #$$apiManagementservice.ListOfAPIRemediated
+            $apiMgmtContext = New-AzAPIManagementContext -ResourceGroupName $_.ResourceGroupName -ServiceName $_.ResourceName
+            $apiMgmt = Get-AzAPIManagementAPI -Context $apiMgmtContext
+            $apiManagementservice.ListOfAPIRemediated  = $apiManagementservice.ListOfAPIRemediated -split ","
+            $apiMgmt | ForEach-Object {      
+                if($apiManagementservice.ListOfAPIRemediated -contains  $_.Name)
+                {
+                    Set-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId -Protocols @("Http") -ErrorAction SilentlyContinue
+                    $output = (Get-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId  -ErrorAction SilentlyContinue).Protocols 
+                        
+                    if($output -eq "Http")
+                    {
+                       
+                        $listOfApiRolledBack += $_.Name
+                    }
+                    else
+                    {
+                        $listOfApiSkipped += $_.Name
+                    }
+                }
+            } 
+
+            if($listOfApiRolledBack.count -ne '0')
+            {
+                $listOfApiRolledBack = $listOfApiRolledBack -join ","
+                $apiManagementServicesRolledBack += $apiManagementservice |Select-Object @{N='ResourceID';E={$_.ResourceId}},
+                                                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                                        @{N='ResourceName';E={$_.ResourceName}},
+                                                                                        @{N='ListOfAPIRolledBack';E={$listOfApiRolledBack}}
+            }
+
+            if($listOfApiSkipped.count -ne '0')
```

Comment:
Even if there is one `skipped` API in an APIM resource, the resource rollback should be treated as an error/incomplete.